### PR TITLE
feat: add C implementation for `math/base/special/fast/max`

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves a part of #649 .

## Description

> What is the purpose of this pull request?

This pull request:

-   add C implementation for math/base/special/fast/max

## Related Issues

> Does this pull request have any related issues?

This pull request:

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/fast/max/examples/c/example.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    double y;
+    double v;
+    int i;
+
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( (double)rand() / (double)RAND_MAX ) * 2000.0 ) - 1000.0;
+        y = ( ( (double)rand() / (double)RAND_MAX ) * 2000.0 ) - 1000.0;
+        v = stdlib_base_fast_max( x, y );
+        printf( "x: %lf, y: %lf, max(x, y): %lf\n", x, y, v );
+    }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/max/examples/c/example.c]:
```suggestion
	double x;
	double y;
	double v;
	int i;

	for ( i = 0; i < 100; i++ ) {
		x = ( ( (double)rand() / (double)RAND_MAX ) * 2000.0 ) - 1000.0;
		y = ( ( (double)rand() / (double)RAND_MAX ) * 2000.0 ) - 1000.0;
		v = stdlib_base_fast_max( x, y );
		printf( "x: %lf, y: %lf, max(x, y): %lf\n", x, y, v );
	}
```

File: lib/node_modules/@stdlib/math/base/special/fast/max/lib/native.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Return the maximum value.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/max/lib/native.js]:
```suggestion
* Returns the maximum value.
```

File: lib/node_modules/@stdlib/math/base/special/fast/max/src/main.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+*
+* @example
+* double v = stdlib_base_fast_max( +0.0, -0.0 );
+* // returns -0.0
+*/
+double stdlib_base_fast_max( const double x, const double y ) {
+    if ( x > y ) {
+        return x;
+    }
+    return y;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/max/src/main.c]:
```suggestion
	if ( x > y ) {
		return x;
	}
	return y;
```

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @Rejoan-Sardar !

File: lib/node_modules/@stdlib/math/base/special/fast/max/lib/native.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+* @param {number} y - second number
+* @returns {number} maximum value
+*
+* @example
+* var v = max( 4.2, 3.14 );
+* // returns 4.2
+*
+* @example
+* var v = max( 0.0, -0.0 );
+* // returns 0.0
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/fast/max/lib/native.js]: Per tests in `test.native.js`, this should be:

```suggestion
* // returns -0.0
```

File: lib/node_modules/@stdlib/math/base/special/fast/max/lib/native.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+*
+* @example
+* var v = max( 3.14, NaN );
+* // returns NaN
+*
+* @example
+* var v = max( NaN, 5.0 );
+* // returns NaN
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/fast/max/lib/native.js]: Per tests in `test.native.js`, this should be:

```suggestion
* var v = max( NaN, 5.0 );
* // returns 5.0
```

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**